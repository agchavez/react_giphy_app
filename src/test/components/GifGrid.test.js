import { shallow } from "enzyme"
import '@testing-library/jest-dom'
import React from "react";
import { GifGrid } from "../../components/GifGrid"
import { useFetchGifs } from "../../hooks/useFetchGifs";
jest.mock('../../hooks/useFetchGifs');


describe('Prueba del <GifGrid/>', () => {
    const categorie = "avengers";
    
    
    test('Debe mostarse el componente normal ', () => {
        useFetchGifs.mockReturnValue({
            data:[],
            loading:true
        });
        const wrapper = shallow(<GifGrid category = {categorie} />);
        expect(wrapper).toMatchSnapshot();
   });

   test('Debe de mostar items cuando se mandan imagenes del useFetchGefs', () => {
        const gifs = [
            {
                id:'ABC',
                url:'htpps://localhost/cuaqluiercosa/asldkd.gif',
                title:'Cualquier cosa'

            },
            {
                id:'ABCD',
                url:'htpps://localhost/cuaqluiercosa/asldkd.gif',
                title:'Cualquier cosa'

            }
        ]
        useFetchGifs.mockReturnValue({
            data:gifs,
            loading:false
        });

        const wrapper = shallow(<GifGrid category = {categorie} />);
        expect(wrapper).toMatchSnapshot();
        expect(wrapper.find('p').exists()).toBe(false);
        expect(wrapper.find('GifGridItem').length).toBe(gifs.length);


       
   })
   
   


    
})
