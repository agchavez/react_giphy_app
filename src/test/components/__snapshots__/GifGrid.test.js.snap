// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Prueba del <GifGrid/> Debe de mostar items cuando se mandan imagenes del useFetchGefs 1`] = `
<Fragment>
  <h3
    className="animate__animated animate__fadeIn"
  >
     
    avengers
     
  </h3>
  <div
    className="card-grid"
  >
    <GifGridItem
      id="ABC"
      key="ABC"
      title="Cualquier cosa"
      url="htpps://localhost/cuaqluiercosa/asldkd.gif"
    />
    <GifGridItem
      id="ABCD"
      key="ABCD"
      title="Cualquier cosa"
      url="htpps://localhost/cuaqluiercosa/asldkd.gif"
    />
  </div>
</Fragment>
`;

exports[`Prueba del <GifGrid/> Debe mostarse el componente normal  1`] = `
<Fragment>
  <h3
    className="animate__animated animate__fadeIn"
  >
     
    avengers
     
  </h3>
  <p
    className="animate__animated animate__flash"
  >
    Loading
  </p>
  <div
    className="card-grid"
  />
</Fragment>
`;
